tool.c

```diff
@@ -303,11 +297,9 @@ targets(int argc, char *argv[])
 	}
 
 	struct edge *e = NULL;
+	size_t n;
 	for (e = alledges; e; e = e->allnext) {
-		if (e->nout == 0)
-			continue;
-
-		for (size_t n = 0; n < e->nout; ++n) {
+		for (n = 0; n < e->nout; ++n) {
 			if (e->out[n]->nuse != 0)
 				continue;
 
```

Comment:
This loop didn't seem quite right to me. For every output that is not used by any other action, `targets_depth` is called for the entire list of outputs.

I think this would cause redundancy if there are multiple outputs, and if there were some that were used by other actions, they would get printed anyway.

I pushed a commit which restructures `targets_depth` slightly, moving the loop to surround the recursive call, instead of looping at the toplevel. I believe it should resolve those issues.